youtube_dl/extractor/iqiyi.py

```diff
@@ -301,10 +302,14 @@ def get_raw_data(self, tvid, video_id):
     def _extract_playlist(self, webpage):
         PAGE_SIZE = 50
 
-        links = re.findall(
-            r'<a[^>]+class="site-piclist_pic_link"[^>]+href="(http://www\.iqiyi\.com/.+\.html)"',
-            webpage)
-        if not links:
+        links = []
+        for link in re.findall(r'<a[^>]+class="[^"]*site-piclist_pic_link[^"]*"[^>]*>', webpage):
```

Comment:
Javascript (HTML DOM API) has ```getElementById``` only. If a webpage has multiple elements with the same ID, website developers have to do parsing stuffs if they want to access all elements of that ID. It would be suprising to me if there's really such a site.

Also, it's better to make function in utils match Javascript.